graphql/exec.go

```diff
@@ -21,12 +21,75 @@ func CollectFields(doc *query.Document, selSet []query.Selection, satisfies []st
 	return collectFields(doc, selSet, satisfies, variables, map[string]bool{})
 }
 
+func runDirectives(directives common.DirectiveList, variables map[string]interface{}) bool {
+	for _, directive := range directives {
+		if !runSingleDirective(directive, variables) {
+			return false
+		}
+	}
+	return true
+}
+
+func runSingleDirective(directive *common.Directive, variables map[string]interface{}) bool {
+	arg := directive.Args.MustGet("if")
+	fieldAllowed := true
+	switch arg.(type) {
+	case *common.Variable:
+		{
+			value := arg.Value(variables)
+			if realValue, ok := value.(bool); ok {
+				fieldAllowed = realValue
+			}
+		}
+	case *common.BasicLit:
+		{
+			value := arg.Value(nil)
+			if realValue, ok := value.(bool); ok {
+				fieldAllowed = realValue
+			}
+		}
+	}
+
+	switch directive.Name.Name {
+	case "skip":
+		{
+			return !fieldAllowed
+		}
+	case "include":
+		{
+			return fieldAllowed
+		}
+	default:
+		// Custom directives are not supported yet
+		return fieldAllowed
+	}
+}
+
+func populateVariables(operation *query.Operation, variables map[string]interface{}) map[string]interface{} {
+	if variables == nil {
+		variables = make(map[string]interface{})
+	}
+	for _, variable := range operation.Vars {
+		if _, ok := variables[variable.Name.Name]; !ok && variable.Default != nil {
+			variables[variable.Name.Name] = variable.Default.Value(nil)
+		}
+	}
+	return variables
+}
+
 func collectFields(doc *query.Document, selSet []query.Selection, satisfies []string, variables map[string]interface{}, visited map[string]bool) []CollectedField {
 	var groupedFields []CollectedField
-
+	oop, err := doc.GetOperation("")
```

Comment:
:worried: 

Maybe these methods should be refactored to take the request context instead of document so they can get the correct operation

Reply:
This comment sill applies, everywhere that calls CollectFields is grabbing the doc out of the RequestContext embedded in execution context, they could pretty easily be updated to do
```go
fields := graphql.CollectFields(ec.RequestContext, sel, implementors)
```
instead of
```go
fields := graphql.CollectFields(ec.Doc, sel, implementors, ec.Variables)
```

and then you have access to the right operation instead of assuming it was an anonymous operation.